packages/nx/src/command-line/import/import.ts

```diff
@@ -391,76 +392,66 @@ async function createTemporaryRemote(
   await destinationGitClient.fetch(remoteName);
 }
 
-// If the user imports a project that isn't in NPM/Yarn/PNPM workspaces, then its dependencies
-// will not be installed. We should warn users and provide instructions on how to fix this.
-async function warnOnMissingWorkspacesEntry(
+/**
+ * If the user imports a project that isn't in the workspaces entry, we should add that path to the workspaces entry.
+ */
+async function handleMissingWorkspacesEntry(
   pm: PackageManager,
-  pmc: PackageManagerCommands,
-  pkgPath: string
+  pkgPath: string,
+  destinationGitClient: GitRepository
 ) {
+  let added = false;
+  try {
+    added = addPackagePathToWorkspaces(pm, workspaceRoot, pkgPath);
+  } catch {
+    return;
+  }
+  if (!added) {
+    return;
+  }
+  await destinationGitClient.amendCommit();
   if (!isWorkspacesEnabled(pm, workspaceRoot)) {
     output.warn({
       title: `Missing workspaces in package.json`,
       bodyLines:
         pm === 'npm'
           ? [
               `We recommend enabling NPM workspaces to install dependencies for the imported project.`,
-              `Add \`"workspaces": ["${pkgPath}"]\` to package.json and run "${pmc.install}".`,
+              `Added \`"workspaces": ["${pkgPath}"]\` to package.json.`,
               `See: https://docs.npmjs.com/cli/using-npm/workspaces`,
             ]
           : pm === 'yarn'
           ? [
               `We recommend enabling Yarn workspaces to install dependencies for the imported project.`,
-              `Add \`"workspaces": ["${pkgPath}"]\` to package.json and run "${pmc.install}".`,
+              `Added \`"workspaces": ["${pkgPath}"]\` to package.json.`,
               `See: https://yarnpkg.com/features/workspaces`,
             ]
           : pm === 'bun'
           ? [
               `We recommend enabling Bun workspaces to install dependencies for the imported project.`,
-              `Add \`"workspaces": ["${pkgPath}"]\` to package.json and run "${pmc.install}".`,
+              `Added \`"workspaces": ["${pkgPath}"]\` to package.json.`,
               `See: https://bun.sh/docs/install/workspaces`,
             ]
           : [
               `We recommend enabling PNPM workspaces to install dependencies for the imported project.`,
-              `Add the following entry to to pnpm-workspace.yaml and run "${pmc.install}":`,
+              `Added the following entry to to pnpm-workspace.yaml:`,
```

Comment:
I don't think this should change. If someone doesn't have `npm workspaces` setup, we shouldn't enable it for them. We should keep it as is and tell them to enable it altogether. If someone does have `npm workspaces` enabled, then that's when we should add it for them.

Reply:
changed back